### ISSUE TITLE
Export-FormatData ... still does not.

### ISSUE THREAD

**Issue by Jaykul** [PowerShell/PowerShell#24332]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Export-FormatData frequently does not export the right information
Export-FormatData does not export all the types in a format definition
Export-FormatData does not export CustomEntry ScriptBlocks

If you look at the result of 
```PowerShell
Get-FormatData -TypeName *ErrorRecord* -ov erv | Export-FormatData -Path OriginalErrorView.format.ps1xml
```

And compare that to `$erv` and `$erv.FormatViewDefinition.Control.Entries.CustomItems` you can see:

1. There should be _two_ types for the first view, but only one is exported.
2. The scriptblocks just come out as `<ScriptBlock>;</ScriptBlock>` 

I'm writing this up **AGAIN** because the idle bot **KEEPS CLOSING** the issues.

https://github.com/PowerShell/PowerShell/issues/9300
https://github.com/PowerShell/PowerShell/issues/11596
https://github.com/PowerShell/PowerShell/issues/11308

### Expected behavior

```console
# This should work:
Get-FormatData -TypeName *Error* -ov erv | Export-FormatData -Path OriginalErrorView.format.ps1xml
Update-FormatData -PrependPath OriginalErrorView.format.ps1xml

# This should produce the same error as normal
stop-service lanmanworkstation
```

### Actual behavior

```console
# After running these commands, no errors will ever be visible again:
Get-FormatData -TypeName *Error* -ov erv | Export-FormatData -Path OriginalErrorView.format.ps1xml
Update-FormatData -PrependPath OriginalErrorView.format.ps1xml

# This produces no visible output
stop-service lanmanworkstation
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.5.0-preview.4
PSEdition                      Core
GitCommitId                    7.5.0-preview.4
OS                             Microsoft Windows 10.0.22635
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by SeeminglyScience** [comment]
One quick note, for the scriptblocks there is a parameter on `Export-FormatData`: `-IncludeScriptBlock`.

Don't know the history or reason behind it, but that's needed for the scriptblocks

**Comment by Jaykul** [comment]
You are correct. That's much better, but still totally broken for many types. 

If you do something like this, for instance:

```PowerShell
Get-FormatData -TypeName System.IO.FileInfo -ov fi | Export-FormatData -Path FileInfo.format.ps1xml -IncludeScriptBlock
```

The resulting format file does not actually contain anything about the FileInfo class, because that was the **second type** in the format data.
